build.rs

```diff
@@ -101,6 +103,38 @@ fn write_postbuild_conf_pri(
     Ok(())
 }
 
+fn check_boost_version() -> Result<bool, Box<dyn Error>> {
+    let output = Command::new("bash")
+    .args(["-c", "grep \"BOOST_LIB_VERSION\" $(find /usr/local/Cellar/boost/ /usr/include /usr/local/include -name 'version.hpp' 2> /dev/null | grep 'boost/version.hpp') | grep -v \"//\" | cut -d '\"' -f2 | sed 's/_/./g'"])
+    .output()?;
+
+    let version_str = String::from_utf8(output.stdout)?.trim().to_string();
+    let version_parts: Vec<&str> = version_str.split('.').collect();
+
+    if version_parts.len() < 2 {
+        return Ok(false);
+    }
+
+    let major_version = version_parts[0].parse::<i32>()?;
+    let minor_version = version_parts[1].parse::<i32>()?;
+
+    Ok(major_version > 1 || (major_version == 1 && minor_version >= 71))
+}
+
+fn find_boost_include_path() -> Option<String> {
+    let possible_paths = vec!["/usr/local/include", "/usr/include"];
+    let boost_header = "boost/signals2.hpp";
```

Comment:
How about searching for version.hpp here instead of signals2.hpp, and then reading the version number out of the found file, rather than doing the version check separately in `check_boost_version` ? That would help ensure the path being version checked is the same path being used.

Also try to use Rust code to parse the file rather than shell commands.

BTW, for checking the version string, you can reuse the `check_version` function.

Reply:
great points! 
I have updated the PR to include one `get_boost_path` fn where boost version is read from version.hpp file and verified by `check_verison` func.